src/v/storage/segment_set.cc

```diff
@@ -129,6 +130,18 @@ segment_set::lower_bound(model::offset offset) const {
 // on that time index to find the closest index entry and scan the log from
 // there. Otherwise it will move on to the next log segment.
 segment_set::iterator segment_set::lower_bound(model::timestamp needle) {
+    // Note that we exclude the segments that only contain configuration batches
+    // from our search, as their timestamps may be wildly different from the
+    // user provided timestamps.
+    return filtered_lower_bound(
+      _handles.begin(),
+      _handles.end(),
+      needle,
+      segment_ordering{},
+      [](const auto& segment) {
+          return segment->index().non_data_timestamps() == false;
+      });
+
     return std::lower_bound(
```

Comment:
Did you mean to remove these two lines?

Reply:
Oops. Must have been lost in the rebase. Will update.

Reply:
Fixed.